src/vs/editor/common/commonCodeEditor.ts

```diff
@@ -622,6 +622,10 @@ export abstract class CommonCodeEditor extends EventEmitter implements editorCom
 		return this.model.deltaDecorations(oldDecorations, newDecorations, this.id);
 	}
 
```

Comment:
setDecorators actually uses deltaDecorators. Inline decorators need to use after.contentText which ends up translating to a css:after block. For each of the unique decorators we need to create the selectors. By not using setDecorators we'll have to do that manually which will duplicate the logic in two places. 

setDecorators ends up calling `provider = new DecorationSubTypeOptionsProvider(this._styleSheet, key, parentTypeKey, options);` eventually.

setDecorators does the correct thing and provides the proper abstraction. Its just that before calling setDecorators, the decoratorTypeKey to be registered. That's why I added the registerDecoratorType api. 

I've spent hours trying to figure out if there was a nicer way to register the decoratorType. This seemed like the least obtrusive way since `this._registerDecorationType` was already being used.

The other workaround I can think is for setDecorators to auto-register the key if it doesn't exist.